src/wms/src/main/java/org/geoserver/wms/legendgraphic/LegendMerger.java

```diff
@@ -0,0 +1,662 @@
+/* (c) 2016 Open Source Geospatial Foundation - all rights reserved
+ * This code is licensed under the GPL 2.0 license, available at the root
+ * application directory.
+ */
+
+package org.geoserver.wms.legendgraphic;
+
+import java.awt.Color;
+import java.awt.Font;
+import java.awt.Graphics2D;
+import java.awt.RenderingHints;
+import java.awt.image.BufferedImage;
+import java.awt.image.IndexColorModel;
+import java.awt.image.RenderedImage;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Locale;
+import java.util.Map;
+
+import org.geoserver.wms.GetLegendGraphicRequest;
+import org.geoserver.wms.legendgraphic.LegendUtils.LegendLayout;
+import org.geoserver.wms.map.ImageUtils;
+import org.geotools.styling.Description;
+import org.geotools.styling.Rule;
+import org.opengis.util.InternationalString;
+
+public class LegendMerger {
+
+    /**
+     * Receives a list of <code>BufferedImages</code> and produces a new one which holds all the images in <code>imageStack</code> one above the
+     * other.
+     * 
+     * @param imageStack the list of BufferedImages
+     * @param dx horizontal space between images
+     * @param dy vertical space between images
+     * @param margin padding around image
+     * @param backgroundColor background color of legend
+     * @param transparent if true make legend transparent
+     * @param antialias if true applies anti aliasing
+     * @param layout orientation of legend, my be horizontal or vertical
+     * @param rowWidth maximum width for horizontal legend
+     * @param rows maximum number of rows for horizontal legend
+     * @param columnHeight maximum height for vertical legend
+     * @param columns maximum number of columns for vertical legend
+     * @return the legend image with all the images on the argument list.
+     */
+    public static BufferedImage mergeRasterLegends(List<BufferedImage> imageStack, int dx, int dy,

```

Comment:
I added a MergeOptions object to contain all the arguments and refactored the code to use it in various places